engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.h

```diff
@@ -136,11 +137,13 @@ class ComputeAggregateKernel : public kernel {
 class DistributeAggregateKernel : public kernel {
 public:
 	DistributeAggregateKernel(std::size_t kernel_id, const std::string & queryString, std::shared_ptr<Context> context, std::shared_ptr<ral::cache::graph> query_graph)
-		: kernel{kernel_id, queryString, context, kernel_type::DistributeAggregateKernel} {
-        this->query_graph = query_graph;
+		: kernel{kernel_id, queryString, context, kernel_type::DistributeAggregateKernel},
+		  message_manager{kernel_id, context, ral::communication::CommunicationData::getInstance().getSelfNode().id(), ral::communication::CommunicationData::getInstance().getSelfNode(),
```

Comment:
Do not instantiate any classes with some global function call. The more we do this the less we can write tests for things. Do this on the outside before calling the constructor.

Reply:
Here it is instantiating the message manager for this particular kernel. It is part of the kernel. So it should be instantiated in the Constructor of the kernel

Reply:
```
ral::communication::CommunicationData::getInstance().getSelfNode().id()

```
im referring to that